Source/WebCore/platform/graphics/MediaPlayer.cpp

```diff
@@ -102,10 +102,13 @@ namespace WebCore {
 
 // a null player to make MediaPlayer logic simpler
 
-class NullMediaPlayerPrivate final : public MediaPlayerPrivateInterface {
+class NullMediaPlayerPrivate final : public MediaPlayerPrivateInterface, public RefCounted<NullMediaPlayerPrivate> {
 public:
     explicit NullMediaPlayerPrivate(MediaPlayer*) { }
```

Comment:
Ref counted classes should never have public constructors. This is too error prone. The constructor should be private and there should be a public create() function to create it.